app/src/main/java/com/example/dogbook/main/fragments/MapFragment.java

```diff
@@ -107,7 +109,13 @@ public void onCameraIdle() {
                 LatLngBounds visibleAreaBox = map.getProjection().getVisibleRegion().latLngBounds;
                 Log.i(TAG, "New visible box: " + visibleAreaBox.toString());
                 if (!isVisibleAreaCovered(visibleAreaBox)) {
-                    updateCoveredAreaBox(visibleAreaBox);
+                    if (updatePostsThread != null && updatePostsThread.isAlive()) {
+                        updatePostsThread.interrupt();
+                    }
+                    updatePostsThread = new Thread(new UpdatePostsRunnable(visibleAreaBox));
+                    updatePostsThread.start();
```

Comment:
Hm, does this actually end up canceling the Parse Query if you have one pending?  I think this might just stop execution on that thread, but if you still add a parse query going on, we wouldn't stop/cancel the query.  We'd just never handle the response?

If this is the case, then I think this code works, but we can likely optimize it a bit more to cancel spurious requests.

The reason to do this is that it ensures we don't spin up a new thread unless we absolutely have to do so (which will be when we go to actually make the parse query, itself, and not really an additional extra thread here to do all of this processing).  Similarly, will the parse query always execute on a thread that it spins up or does it only move it to a new thread if it's called from the main thread?  If the latter is the case, then it at least means we're not creating more threads than before

Can we update this logic such that:
- We try to cancel the actual ongoing parse query itself if it's ongoing
- Then spin up a new query?